e2e/nomostest/testpredicates/predicates.go

```diff
@@ -1171,3 +1171,51 @@ func RootSyncSpecOverrideEquals(expected *v1beta1.RootSyncOverrideSpec) Predicat
 		return nil
 	}
 }
+
+func subjectsEqual(want []string, got []rbacv1.Subject) error {
+	if len(got) != len(want) {
+		return errors.Wrapf(ErrFailedPredicate, "want %v subjects; got %v", want, got)
```

Comment:
This ErrFailedPredicate pattern seems weird. I would expect the caller of the predicates to add it as a wrapper, rather that every individual predicate to need to return it, especially as the root cause...